### ISSUE TITLE
use .pth trained by myself to run eval.py is too slow

### ISSUE THREAD

**Issue by hust-kevin** [amdegroot/ssd.pytorch#215]
when I use the ssd300_mAP_77.43_v2.pth to eval, 100 picture only spend 3.5s

im_detect: 100/4952 3.978s
im_detect: 200/4952 7.211s

but after I run train.py saved the weight, and I use the weight trained by myself is very slow, 100 picture spend 35s

im_detect: 100/4952 34.390s
im_detect: 200/4952 67.907s
im_detect: 300/4952 101.621s
im_detect: 400/4952 135.185s
im_detect: 500/4952 168.701s
im_detect: 600/4952 201.940s

**Comment by hust-kevin** [comment]
@amdegroot  do you know the reason

**Comment by TheShadow29** [comment]
I am also facing a similar issue. @hust-kevin were you able to solve it?

**Comment by hust-kevin** [comment]
@TheShadow29  I found that after train more iteration，detection will be more quickly

**Comment by TheShadow29** [comment]
Strange, for me there was no improvement. I also opened a discussion here https://discuss.pytorch.org/t/getting-differing-speeds-on-forward-pass-with-different-weights/26628?u=theshadow29

**Comment by TheShadow29** [comment]
It turns out you were right. So after more iterations, the num of predicted boxes drastically reduces allowing efficient computation of nms and hence the faster speed for evaluation

**Comment by hust-kevin** [comment]
@TheShadow29 I think so

**Comment by pengfeidip** [comment]
actually in  [eval.py](https://github.com/amdegroot/ssd.pytorch/blob/master/eval.py) file, line 391, the code
 ```     python 
        mask = dets[:, 0].gt(0.).expand(5, dets.size(0)).t()
 ```
` .gt(0.)` means  the bounding boxes will be parsed  if it's confidence is greater then 0.0. So, the parameter `--confidence_threshold`  that user inputs is not used

**Comment by pengfeidip** [comment]
> actually in [eval.py](https://github.com/amdegroot/ssd.pytorch/blob/master/eval.py) file, line 391, the code
> 
> ```python
>        mask = dets[:, 0].gt(0.).expand(5, dets.size(0)).t()
> ```
> 
> ` .gt(0.)` means the bounding boxes will be parsed if it's confidence is greater then 0.0. So, the parameter `--confidence_threshold` that user inputs is not used

I was wrong, please ignore me (*￣︶￣)

**Comment by lianxxx** [comment]
> > actually in [eval.py](https://github.com/amdegroot/ssd.pytorch/blob/master/eval.py) file, line 391, the code
> > ```python
> >        mask = dets[:, 0].gt(0.).expand(5, dets.size(0)).t()
> > ```
> > 
> > 
> > ` .gt(0.)` means the bounding boxes will be parsed if it's confidence is greater then 0.0. So, the parameter `--confidence_threshold` that user inputs is not used
> 
> I was wrong, please ignore me (*￣︶￣)

But I am confused dets[:,0] should be already greater than 0, why dets[:,0].gt(0.) can filter some b-boxes out?

**Comment by krxat** [comment]
@TheShadow29 Hi, did you train on your own dataset?
If so, can you please explain the steps you took?
Also how did you solve the timing issue??
Thanks

**Comment by TheShadow29** [comment]
@cypherix I didn't train on my dataset, but the easiest way would be to convert your dataset into coco format. The timing issue happens due to large number of boxes in the validation step and the subsequent nms makes it too slow. The easiest way is to simply validate after a few epochs, the number of boxes would be reduced a lot and nms should be fast.

**Comment by cypherix** [comment]
@TheShadow29 How many iterations did you train for and was it working with good accuracy??

**Comment by leijuzi** [comment]
![image](https://user-images.githubusercontent.com/42333539/57276745-138f1580-70d5-11e9-9a48-7593a9248e0e.png)
hi this is my weight trained by my own datasets ,voc format ,i dont know why my weights size is 103733, i did not change the net, could u help me,thx
